src/setup.py

```diff
@@ -305,20 +306,36 @@ def __init__(self, credentials_path=None, access_key=None, secret_key=None, regi
 class InteractiveConfigurator(object):
     DEFAULT_PROMPT = "Enter choice [" + Color.green("{default}") + "]: "
 
-    def __init__(self, plugin_config, metadata_reader, collectd_info):
+    def __init__(self, plugin_config, metadata_reader, collectd_info, non_interactive, region, host,
+                 proxy_name, proxy_port, access_key, secret_key, creds_path):
         self.config = plugin_config
         self.metadata_reader = metadata_reader
         self.collectd_info = collectd_info
+        self.non_interactive = non_interactive
+        self.region = region
+        self.host = host
+        self.proxy_name = proxy_name
+        self.proxy_port = proxy_port
+        self.access_key = access_key
+        self.secret_key = secret_key
+        self.creds_path = creds_path
 
     def run(self):
-        self._configure_region()
-        self._configure_hostname()
-        self._configure_credentials()
-        self._configure_proxy_server_name()
-        self._configure_proxy_server_port()
-        self._configure_push_asg()
-        self._configure_push_constant()
-        self._configure_plugin_installation_method()
+        if self.non_interactive:
+            self._configure_region_non_interactive()
+            self._configure_hostname_non_interactive()
+            self._configure_credentials_non_interactive()
+            self._configure_proxy_server_name_non_interactive()
+            self._configure_proxy_server_port_non_interactive()
+        else:
+            self._configure_region()
+            self._configure_hostname()
+            self._configure_credentials()
+            self._configure_proxy_server_name()
+            self._configure_proxy_server_port()
+            self._configure_push_asg()
+            self._configure_push_constant()
```

Comment:
Nice to hear that!!! So, my goal is to do the same functionality as interactive setup.   self._configure_proxy_server_port() and  self._configure_proxy_server_name() already added, working on self._configure_push_asg() and self._configure_push_constant() now.